### PR TITLE
feat(developer): Add note about `IFilenameValidator`

### PR DESCRIPTION
**Pull request by susnux**
### ☑️ Resolves

* For https://github.com/nextcloud/server/pull/46538
* And for https://github.com/nextcloud/server/pull/46371

### 🖼️ Screenshots

Screenshot 2024-07-17 at 14-14-41 Upgrade to Nextcloud 30 — Nextcloud latest Developer Manual latest documentation

### CODE REVIEW

File: developer_manual/app_publishing_maintenance/app_upgrade_guide/upgrade_to_30.rst [hunk 1]
```diff
@@ -100,6 +100,7 @@ Added APIs
 - ``OCA\Files_Trashbin\Trash\ITrashItem::getDeletedBy`` should return the user who deleted the item or null if unknown
 - ``OCP\IUser::getPasswordHash`` should return the password hash of the user
 - ``OCP\IUser::setPasswordHash`` should set the password hash of the user
+- ``OCP\Files\IFilenameValidator`` was added to allow storage independen filename validation.
```
Comment (Pytal) [inline developer_manual/app_publishing_maintenance/app_upgrade_guide/upgrade_to_30.rst:103]:
```suggestion
- ``OCP\Files\IFilenameValidator`` was added to allow storage independent filename validation.
```

File: developer_manual/app_publishing_maintenance/app_upgrade_guide/upgrade_to_30.rst [hunk 2]
```diff
@@ -124,5 +125,7 @@ Changed APIs
 Removed APIs
 ^^^^^^^^^^^^
 
+- ``OCP\Util::isValidFileName`` was deprecated in 8.1.0 and is now removed, use either ``OCP\Files\Storage\IStorage::verifyPath`` or the new ``OCP\Files\IFilenameValidator``.
+
```
Comment (Pytal) [inline developer_manual/app_publishing_maintenance/app_upgrade_guide/upgrade_to_30.rst:129]: `getForbiddenFileNameChars` was also removed from OCP\Util in the PR?

Comment (susnux) [reply developer_manual/app_publishing_maintenance/app_upgrade_guide/upgrade_to_30.rst:145]: done

File: developer_manual/app_publishing_maintenance/app_upgrade_guide/upgrade_to_30.rst (hunk 1 above)
Comment (Pytal) [reply developer_manual/app_publishing_maintenance/app_upgrade_guide/upgrade_to_30.rst:105]: Wasn't fixed?

**Review by Pytal — APPROVED** [review]

**Pull request merged**